xds/internal/balancer/edsbalancer/eds_impl.go

```diff
@@ -277,9 +277,15 @@ func (edsImpl *edsBalancerImpl) handleEDSResponsePerPriority(bgwc *balancerGroup
 				Addr: lbEndpoint.Address,
 			}
 			if edsImpl.subBalancerBuilder.Name() == weightedroundrobin.Name && lbEndpoint.Weight != 0 {
-				address.Metadata = &weightedroundrobin.AddrInfo{
-					Weight: lbEndpoint.Weight,
-				}
+				ai := weightedroundrobin.AddrInfo{Weight: lbEndpoint.Weight}
+				weightedroundrobin.SetAddrInfo(ai, &address)
+				// Metadata field in resolver.Address is deprecated. The
+				// attributes field should be used to specify arbitrary
+				// attributes about the address. We still need to populate the
+				// Metadata field here to allow users of this field to migrate
+				// to the new one.
+				// TODO(easwars): Remove this once all users have migrated.
```

Comment:
Is this behavior documented?  Who are the users you're referring to?  Only us?

Reply:
@menghanl mentioned that dropbox was a user.

Reply:
I'm a little lost.  This is the address we're passing to the weighted round robin balancer - why do we need to set Metadata for it?  Are they using xds and overriding our weighted RR balancer?

Reply:
I'm not very sure how (or for what) they use it. @menghanl might have an answer.

Reply:
Yes. And they use edf wrr #2968

Reply:
@menghanl How will we track the safe removal of this?

Reply:
Keep it for one more release, and remove after 1.30?

Reply:
That's fine with me if it's OK with dropbox.  Please file an issue and cc them?

Reply:
Issue filed #3563